File: src/main/java/com/yama331/restapi_crud/entity/Consultant.java
package com.yama331.restapi_crud.entity;

import java.time.LocalDate;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;

@Entity // 이 클래스는 JPA가 관리하는 테이블의 짝이다.
@Table(name="consultants") // 실제 테이블 이름. 자바 클래스 이름과 다를 수 있어서 명시하는 거임.
@lombok.Getter // 
@lombok.Setter // Setter가 없으면 수정이 불가능.
@NoArgsConstructor // 인자가 없는 생성자 JPA필수, 
@AllArgsConstructor // 모든 필드를 한번에 받는 생성자.ㄴ
@Builder // Consultant.builder().name("...").build() 형태로 생성 가능 , 레고처럼 필요한 칸만 골라 조립. create가 이 방식으로 엔티티를 만든다. id와 crated_at등은 조립에 안넣는다. DB가채운다.
public class Consultant {

    @Id // 이 필드가 기본키 라는것.
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(name = "employee_code", nullable = false, length = 20)
    private String employeeCode;

    @Column(nullable = false, length = 20)
    private String phone;

    @Column(length = 100)
    private String email;

    @Column(name = "hire_date")
    private LocalDate hireDate;

    @Column(nullable = false, length = 20)
    private String status = "재직";

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/consultant.ts
/**
 * 설계사 관련 TypeScript 타입
 *
 * 백엔드 DTO 와 "필드 이름"을 맞춘다.
 * - Request  : 등록/수정 때 보내는 몸통
 * - Response : 서버가 돌려주는 몸통
 *
 * [Java ↔ TypeScript 대응]
 * - String          → string
 * - Long / 숫자     → number
 * - LocalDate       → string  ("YYYY-MM-DD")  ※ JSON 에서는 문자열
 * - LocalDateTime   → string  (ISO 날짜시간 문자열)
 * - null 가능 필드  → string | null 또는 선택 속성 ?
 */

/**
 * 등록/수정 요청 타입
 * = 백엔드 ConsultantRequest(DTO)
 */
export interface ConsultantRequest {
    
    name: string;
    employeeCode: string;
    phone: string;
    email?: string;
    hireDate?: string;
    status: string;

}

/**
 * 서버 응답 타입
 * = 백엔드 ConsultantResponse
 */

export interface ConsultantResponse {
    id: number;
    name: string;
    employeeCode: string;
    phone: string;
    email: string | null;
    hireDate: string | null;
    status: string;
    createdAt: string;
    updatedAt: string;

}